emulator.c

```diff
@@ -39,6 +39,12 @@ execute_instruction(i8080 *cpu, uint8_t opcode)
       }
     case 0x09: // NOLINT
       printf("DAD B");
+      uint32_t bc = (cpu->b << 8) | (cpu->c); // NOLINT
+      uint32_t hl = (cpu->h << 8) | (cpu->l); // NOLINT
+      uint32_t result = hl + bc;
+      update_carry_flag(cpu, result > 0xFF); // NOLINT
```

Comment:
I might be wrong, but shouldn't the carry check for result > 0xFFFF? You want to see if the 16-bit value carries over? 0xFF is only 8-bits.

Reply:
nope you are not wrong i accidentally kept the same logic i used for checking a carry for adding two 8 bit registers instead of updating for two 16 bit register pairs